packages/wonder-blocks-data/util/response-cache.test.js

```diff
@@ -111,117 +112,597 @@ describe("./response-cache.js", () => {
+        describe("no custom cache", () => {
+            it("should add the entry to the cache", () => {
+                // Arrange
+                const cache = new ResponseCache();
+                const fakeHandler: IRequestHandler<string, string> = {
+                    getKey: () => "MY_KEY",
+                    type: "MY_HANDLER",
+                    shouldRefreshCache: () => false,
+                    fulfillRequest: jest.fn(),
+                    cache: null,
+                };
+
+                // Act
+                cache.cacheData(fakeHandler, "options", "data");
+                const result = cache.getEntry(fakeHandler, "options");
+
+                // Assert
+                expect(result).toStrictEqual({data: "data"});
+            });
+
+            it("should replace the entry in the handler subcache", () => {
+                // Arrange
+                const cache = new ResponseCache({
+                    MY_HANDLER: {
+                        MY_KEY: {error: "Oh no!"},
+                    },
+                });
+                const fakeHandler: IRequestHandler<string, string> = {
+                    getKey: () => "MY_KEY",
+                    type: "MY_HANDLER",
+                    shouldRefreshCache: () => false,
+                    fulfillRequest: jest.fn(),
+                    cache: null,
+                };
+
+                // Act
+                cache.cacheData(fakeHandler, "options", "other_data");
+                const result = cache.getEntry(fakeHandler, "options");
+
+                // Assert
+                expect(result).toStrictEqual({data: "other_data"});
+            });
+        });
+
+        describe("with custom cache", () => {
+            it("SSR should use the framework cache, not the custom cache", () => {
+                // Arrange
+                jest.spyOn(Server, "isServerSide").mockReturnValue(true);
+                const cache = new ResponseCache();
+                const customCache = {
+                    store: jest.fn(),
+                    retrieve: jest.fn(),
+                    remove: jest.fn(),
+                    removeAll: jest.fn(),
+                };
+                const fakeHandler: IRequestHandler<string, string> = {
+                    getKey: () => "MY_KEY",
+                    type: "MY_HANDLER",
+                    shouldRefreshCache: () => false,
+                    fulfillRequest: jest.fn(),
+                    cache: customCache,
+                };
+
+                // Act
+                cache.cacheData(fakeHandler, "options", "data");
+                /**
+                 * We want to bypass the custom stuff and grab from the real
+                 * storage so we can validate without side-effects.
+                 *
+                 * TODO(somewhatabstract): Isolate framework cache.
+                 * We should probably isolate the framework cache itself better
+                 * to fix this.
```

Comment:
Is this handled in a following PR?

Reply:
Probably. I cannot remember.

Reply:
https://github.com/Khan/wonder-blocks/pull/748#discussion_r377728479